ddtrace/llmobs/_utils.py

```diff
@@ -473,3 +476,137 @@ def on_openai_agent_span_finish(self) -> None:
+
+    pass
+
+
+class LLMObsExportSpansClient:
+    """Retrieves spans from the LLMObs Export API."""
+
+    ENDPOINT = "/api/v2/llm-obs/v1/spans/events"
+    TIMEOUT = 10.0
+    DEFAULT_PAGE_LIMIT = 100
+
+    def __init__(
+        self,
+        api_key: str,
+        app_key: str,
+        site: str,
+    ) -> None:
+        self._api_key: str = api_key
+        self._app_key: str = app_key
+        self._site: str = site
+        self._base_url: str = f"https://api.{self._site}"
+
+    def export_spans(
+        self,
+        span_id: Optional[str] = None,
+        trace_id: Optional[str] = None,
+        tags: Optional[Dict[str, str]] = None,
+        span_kind: Optional[str] = None,
+        span_name: Optional[str] = None,
+        ml_app: Optional[str] = None,
+        from_timestamp: Optional[str] = None,
+        to_timestamp: Optional[str] = None,
+    ):
+        """
+        Generator that yields spans from the LLMObs Export API, handling pagination automatically.
+        """
+        url_options = self._build_url_options(
+            span_id, trace_id, tags, span_kind, span_name, ml_app, from_timestamp, to_timestamp
+        )
+
+        has_next_page = True
+        page_num = 0
+
+        while has_next_page:
+            path = f"{self.ENDPOINT}?{urllib.parse.urlencode(url_options)}"
+            log.debug("Making request to path=%s for page %d", path, page_num)
+
+            resp = self._request(path, self.TIMEOUT)
+
+            if resp.status != 200:
+                log.error(
+                    "Failed to export spans: page=%d, status=%d, response=%s",
+                    page_num,
+                    getattr(resp, "status", None),
+                    getattr(resp, "body", None),
+                )
+                raise ExportSpansAPIError(
+                    f"Failed to export spans with status {resp.status} for url options {url_options}"
+                )
+
+            response_data = resp.get_json() if hasattr(resp, "get_json") else {}
+
+            for span in response_data.get("data", []):
+                yield span.get("attributes", {})
+
+            has_next_page = False
+            next_cursor = self._extract_next_cursor(response_data)
+            if next_cursor:
+                has_next_page = True
+                url_options["page[cursor]"] = next_cursor
+                page_num += 1
+
+    def _request(self, path: str, timeout: float) -> Response:
+        if not self._api_key or not self._app_key:
+            raise ValueError("Both an API key and an APP key are required to make requests to the LLMObs Export API")
+
+        headers = {
+            "Content-Type": "application/vnd.api+json",
+            "DD-API-KEY": self._api_key,
+            "DD-APPLICATION-KEY": self._app_key,
+        }
+
+        conn = get_connection(url=self._base_url, timeout=timeout)
+        try:
+            url = self._base_url + path
+            log.debug("Making GET request to %s", url)
```

Comment:
i think this log statement is duped with the 
```python
 log.debug("Making request to path=%s for page %d", path, page_num)
```
log above, but correct me if i'm wrong!